### PR TITLE
Puppet8x compatibility

### PR DESCRIPTION
**Pull request by flepoutre**
MAJ for puppet8x compatibility

### CODE REVIEW

**Review by flepoutre — COMMENTED** [review]
MAJ for puppet8x compatibility

File: lib/facter/warranty.rb [hunk 1]
```diff
@@ -120,7 +120,7 @@ def create_lenovo_warranty_cache(cache)
     end
 
     # refresh cache daily
-    if File.exists?(cache_file) and Time.now < File.stat(cache_file).mtime + 86400 * 1
```
Comment (JGodin-C2C) [inline lib/facter/warranty.rb:123]: This seems to have been fixed in puppet 8.3.0.

https://www.puppet.com/docs/puppet/8/release_notes_puppet.html#enhancements_puppet_x-8-3-0-PUP-11945

File: manifests/hwtools.pp [hunk 2]
```diff
@@ -6,17 +6,18 @@
 # $dell_repo: use the dell repo for yumrepo, or a already defined one.
 #  The yumrepo should have the name 'dell-omsa-indep'
 #
-class dell::hwtools(
+# @param dell_repo
```
Comment (JGodin-C2C) [inline manifests/hwtools.pp:9]: Definition is already on line 6.

File: manifests/hwtools/package.pp [hunk 3]
```diff
@@ -17,21 +17,21 @@
       # preventing the proper upgrade. To fix it we need to
       # 'yum erase smbios-utils-python python-smbios'
       package { [
-        'smbios-utils-python',
-        'python-smbios',
-      ]:
-        ensure => purged,
-        before => Package['dell-system-update'],
+          'smbios-utils-python',
```
Comment (JGodin-C2C) [inline manifests/hwtools/package.pp:20]: This is relative to formating, not a puppet 8 issue. 
Please open another PR for this .

**Review by JGodin-C2C — COMMENTED** [review]
There is a mix of puppet8 patch and formating 
Please open different PR for different problems.

**Comment by flepoutre** [comment]
Hi,
Thanks for your feedback but is it really necessary to open multiple PR ?

**Comment by JGodin-C2C** [comment]
Yes, this will help readability and bookkeeping. 
Also, i would like to ephasis on the fact that it seems that puppet fixed the exists<>exist problem.

**Comment by JGodin-C2C** [comment]
>  For the fix "exists", I find it a shame to keep the old syntax.

Agreed, it was just a heads up.

**Pull request closed without merging**